api/v1alpha1/component_types.go

```diff
@@ -18,147 +18,70 @@ package v1alpha1
 
 import (
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GitSource describes the Component source
 type GitSource struct {
 	// If importing from git, the repository to create the component from
 	URL string `json:"url"`
 
 	// Specify a branch/tag/commit id. If not specified, default is `main`/`master`.
-	Revision string `json:"revision,omitempty"`
+	//Revision string `json:"revision,omitempty"`
```

Comment:
I think we might need to keep this

Reply:
These are not used anywhere in the source so far.

Reply:
It is used in the `application-service` repo which will eventually rely on this code (see https://github.com/redhat-appstudio/application-service/blob/651fc2e107c413f88dc3796b581ef13e3387243c/controllers/component_controller.go#L210)

Reply:
That is application-service's original types. Note that a duplicate of these types (structs) were made in the common library to make the transition to the library easier. This duplicate or the 'mapping' now needs to be reduced because many of the fields are not necessary when generating the k8s resources.   If someone you need any field added to the 'mapping', then that field must be used by the library in some way.  Currently, what you see now in this PR are the fields that are used.